### DISCUSSION TITLE
Using the usb port instead of hacking the board.

### DISCUSSION THREAD

**Discussion by JugensVan** [luc-github/ESP3DLib#37] [Ideas]
When using the ESP3D lib to communicate with 3D printer via WiFi using a standard ESP32 it is necessary to hack the printer motherboard to get access to the serial port on the other side of the USB chip. Will it be possible to connect it directly to the USB port if you use the ESP32 S2 ??

**Comment by luc-github** [comment]
ESP3DLib does not communicate with printer using serial as it is part of Marlin code itself as a library
ESP32-S2 is one core so I do not recommend to use it hosting Marlin and Wifi at once on one core, also S2 is only supported in esp32-arduino 2.X which is not yet supported by PIO https://github.com/platformio/platform-espressif32/issues/619

I think you mixed ESP3DLib usage which is supposed to be used on esp32 board hosting Marlin and esp3d FW which is a on esp32 connected to printer board by serial,  to connect esp32-s2 to a board using usb you need usb host on one of the board but the esp32-s2 has not such feature - it has otg / and block device usb host but cannot handle serial communication as usb host for all possible printers drivers - this is a little difference but big usage difference

**Reply by JugensVan** [reply]
I only want the esp32 s2 to communicate with the Marlin motherboard via the USB instead of bypassing it and connecting it directly to the serial port on the other side of the USB chip.  I was under the impression that the esp USB port was able to be a host.

**Reply by JugensVan** [reply]
https://docs.espressif.com/projects/esp-idf/en/latest/esp32s2/api-reference/peripherals/usb_host.html
